packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -254,6 +258,22 @@ export class TdsDropdown {
       });
   };
 
+  selectChildrenAsSelectedBasedOnSelectionProp() {
```

Comment:
Don't know if better performance but maybe an enhancement in readability?
```suggestion
  selectChildrenAsSelectedBasedOnSelectionProp() {
  this.children.forEach((element: HTMLTdsDropdownOptionElement) => {
    const isSelected = this.selection.some((selection) => element.value === selection.value);
    element.setSelected(isSelected && (this.multiselect || this.selection.length === 1));
  });
}
```

Reply:
We need to get the children first, because the children might have changed in runtime